packages/studio-ui/src/web/components/Layout/StatusBar/TrainingStatus/index.tsx

```diff
@@ -1,32 +1,95 @@
+import axios from 'axios'
 import { NLU } from 'botpress/sdk'
-import React, { FC } from 'react'
+import { lang } from 'botpress/shared'
+import { Dictionary } from 'lodash'
+import React, { FC, useEffect, useReducer } from 'react'
 import { connect } from 'react-redux'
 import { RootReducer } from '~/reducers'
+import EventBus from '~/util/EventBus'
 
 import MultiLang from './MultiLang'
 import SingleLang from './SingleLang'
 
 interface Props {
   languages: string[]
-  trainSessions: { [lang: string]: NLU.TrainingSession }
 }
 
+interface TrainStatusEvent {
+  botId: string
+  type: 'nlu'
+  trainSession: NLU.TrainingSession
+}
+
+const trainStatusReducer = (state, action) => {
+  if (action.type === 'setAllTrainSessions') {
+    return action.data
+  } else if (action.type === 'setTrainSession') {
+    const trainSession: NLU.TrainingSession = action.data.trainSession
+    return { ...state, [trainSession.language]: trainSession }
+  }
+}
+
+const isTrainStatusEvent = (event: any | TrainStatusEvent): event is TrainStatusEvent =>
+  event.type === 'nlu' && event.trainSession
+
 const TrainingStatusComponent: FC<Props> = (props: Props) => {
-  const { languages, trainSessions } = props
+  const { languages } = props
+  const [trainSessionsState, dispatch] = useReducer(trainStatusReducer, {})
 
-  if (!languages || languages.length < 1) {
-    return null
+  const onStatusBarEvent = (event: any) => {
+    if (isTrainStatusEvent(event) && event.botId === window.BOT_ID) {
+      dispatch({ type: 'setTrainSession', data: { trainSession: event.trainSession } })
+    }
   }
 
-  if (languages.length === 1) {
-    return <SingleLang trainSession={trainSessions[languages[0]]} />
+  const fetchTrainingSessionForLang = async (language: string) => {
```

Comment:
Can you add the return type here? I know that it's inferred, but adding it would make sure that we only expect this type to be returned.